src/OmenAgentResultMapping.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+/// This contract allows for mapping market addresses to IPFS hashes containing agent results.
+contract OmenAgentResultMapping is AccessControlDefaultAdminRules {
+    bytes32 public constant AGENT_ROLE = keccak256("AGENT_ROLE");
+
+    event AgentResultAdded(address indexed marketAddress, bytes32 ipfsHash, address indexed changer);
+
+    // Mapping of a market address to an array of IPFS hashes (bytes32).
+    mapping(address => bytes32[]) private marketToIPFSHashes;
+
+    // We set the deployer as the default admin.
+    constructor()
+        AccessControlDefaultAdminRules(
```

Comment:
Does this mean that only contract creators can add hashes? How do people outside of Gnosis (users of PMAT/PMA, running their own trader or task agents) contribute, then?

Reply:
We need some kind of whitelisting to avoid spam. This way, only the contract creator can add agents to a role.
However, agents made admin can add other agents.

There are many ways to prevent spam, so if this sounds too simplistic, happy to brainstorm

Reply:
Yeah, it's not realistic to expect anyone on the internet to ask us for permission to give us information they had to pay for. I think we will be lucky if people actually don't turn it off in their agents 😄 

> to avoid spam

Even with this permission system, I could ask you to make me an admin, and then I could upload IPFS hashes meant for other trades (not mine).

What we need here is that only the trade maker can upload the hash for his own bets. That way, the agent needs to first place the bet and then upload the information, so it will prevent spam as a secondary behavior.

Reply:
I would be be happy with something like
```
mapping(market_address -> mapping (better_address -> list[Prediction]))
```
Thoughts?

Reply:
object should be 
```
struct Prediction
  ipfsHash:  bytes32
  txHash: bytes32
```

Reply:
Please see newest changes - I believe it now addresses the whitelisting problem (agent can only publish its own predictions). It's also clear which trades correspond to which prediction since both are part of the `Prediction` struct.